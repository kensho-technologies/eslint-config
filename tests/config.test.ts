import path from 'node:path'

import {test, expect} from 'vitest'
import {ESLint} from 'eslint'

test('lints all fixtures', async () => {
  const eslint = new ESLint({ignore: false})
  const results = await eslint.lintFiles([`${import.meta.dirname}/fixtures`])
  results.forEach((result) => {
    const basename = path.basename(result.filePath)
    const messages = result.messages.map((message) => ({
      line: message.line,
      column: message.column,
      message: message.message,
      rule: message.ruleId,
      severity: message.severity,
    }))
    expect(messages).toMatchSnapshot(basename)
  })
})
