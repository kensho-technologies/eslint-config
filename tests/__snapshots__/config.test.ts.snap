// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`lints all fixtures > class.tsx 1`] = `
[
  {
    "column": 7,
    "line": 1,
    "message": "'Foo' is defined but never used.",
    "rule": "@typescript-eslint/no-unused-vars",
    "severity": 2,
  },
]
`;

exports[`lints all fixtures > component.js 1`] = `[]`;

exports[`lints all fixtures > experimental.tsx 1`] = `
[
  {
    "column": 7,
    "line": 4,
    "message": "'nullishCoalescing' is assigned a value but never used.",
    "rule": "@typescript-eslint/no-unused-vars",
    "severity": 2,
  },
  {
    "column": 7,
    "line": 5,
    "message": "'optionalChaining' is assigned a value but never used.",
    "rule": "@typescript-eslint/no-unused-vars",
    "severity": 2,
  },
]
`;

exports[`lints all fixtures > functionComponent.tsx 1`] = `[]`;

exports[`lints all fixtures > import.js 1`] = `[]`;

exports[`lints all fixtures > import.ts 1`] = `[]`;

exports[`lints all fixtures > import-ts-from-js.js 1`] = `[]`;

exports[`lints all fixtures > js-exports.js 1`] = `[]`;

exports[`lints all fixtures > jsdoc.ts 1`] = `
[
  {
    "column": 10,
    "line": 10,
    "message": "'add' is defined but never used.",
    "rule": "@typescript-eslint/no-unused-vars",
    "severity": 2,
  },
  {
    "column": 10,
    "line": 17,
    "message": "'subtract' is defined but never used.",
    "rule": "@typescript-eslint/no-unused-vars",
    "severity": 2,
  },
  {
    "column": 10,
    "line": 21,
    "message": "'multiply' is defined but never used.",
    "rule": "@typescript-eslint/no-unused-vars",
    "severity": 2,
  },
  {
    "column": 10,
    "line": 26,
    "message": "'divide' is defined but never used.",
    "rule": "@typescript-eslint/no-unused-vars",
    "severity": 2,
  },
]
`;

exports[`lints all fixtures > syntax.tsx 1`] = `[]`;

exports[`lints all fixtures > ts-exports.ts 1`] = `[]`;

exports[`lints all fixtures > variables.js 1`] = `
[
  {
    "column": 24,
    "line": 1,
    "message": "'find' is not defined.",
    "rule": "no-undef",
    "severity": 2,
  },
  {
    "column": 24,
    "line": 2,
    "message": "'implicitGlobal' is not defined.",
    "rule": "no-undef",
    "severity": 2,
  },
  {
    "column": 7,
    "line": 3,
    "message": "'unusedA' is assigned a value but never used.",
    "rule": "@typescript-eslint/no-unused-vars",
    "severity": 2,
  },
]
`;

exports[`lints all fixtures > variables.ts 1`] = `
[
  {
    "column": 7,
    "line": 3,
    "message": "'unusedA' is assigned a value but never used.",
    "rule": "@typescript-eslint/no-unused-vars",
    "severity": 2,
  },
  {
    "column": 64,
    "line": 10,
    "message": "'c' is defined but never used.",
    "rule": "@typescript-eslint/no-unused-vars",
    "severity": 2,
  },
]
`;
